delta/plugins/elasticsearch/src/main/scala/ch/epfl/bluebrain/nexus/delta/plugins/elasticsearch/model/ElasticSearchViewCommand.scala

```diff
@@ -0,0 +1,101 @@
+package ch.epfl.bluebrain.nexus.delta.plugins.elasticsearch.model
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.model.Label
+import ch.epfl.bluebrain.nexus.delta.sdk.model.identities.Identity.Subject
+import ch.epfl.bluebrain.nexus.delta.sdk.model.projects.ProjectRef
+import io.circe.Json
+
+/**
+  * Enumeration of ElasticSearch view commands.
+  */
+sealed trait ElasticSearchViewCommand extends Product with Serializable {
+
+  /**
+    * @return the view id
+    */
+  def id: Iri
+
+  /**
+    * @return a reference to the parent project
+    */
+  def project: ProjectRef
+
+  /**
+    * @return the identity associated with this command
+    */
+  def subject: Subject
+}
+
+object ElasticSearchViewCommand {
+
+  /**
+    * Command for the creation of a new ElasticSearch view.
+    *
+    * @param id      the view id
+    * @param project a reference to the parent project
+    * @param value   the view configuration
+    * @param source  the original json value provided by the caller
+    * @param subject the identity associated with this command
+    */
+  final case class CreateElasticSearchView(
+      id: Iri,
+      project: ProjectRef,
+      value: ElasticSearchViewValue,
```

Comment:
shouldn't the command contain just the Json and inside the state machine you should try to convert this json to a `ElasticSearchViewValue`?

This will simplify the routes, having them just get a `entity(as[Json])` and the rest of the logic can be done in the state machine

Reply:
We could put the json in the value no ?

Reply:
it's simple to return back to the client the deserialization errors directly from the routes; we could put the json directly in the value, but it will generate slightly more noise in its use; I would keep it as is.